Automattic/wp-calypso
# Stats: Add Upwork referral banner test

#### Changes proposed in this Pull Request

This PR adds a banner to the Stats page that refers users to Upwork. 

90% of users will still be seeing the default Google My Business banner:
![screenshot 2018-12-18 15 23 25](https://user-images.githubusercontent.com/4924246/50189122-e1714880-02d8-11e9-82de-afe3599e31ae.png)

10% will be seeing the Upwork referral banner:
![screenshot 2018-12-18 15 21 37](https://user-images.githubusercontent.com/4924246/50189084-be469900-02d8-11e9-9c57-e6ca6d1f4b0c.png)

#### Testing instructions

1. Goto to a site: https://wordpress.com/stats/day/belcherjonathan.blog
2 Set variation (`builderReferralBanner`): In console run`localStorage.setItem( 'ABTests', '{"builderReferralStatsNudge_20181218":"builderReferralBanner"}' );`
3. Check if you are seeing the Upwork referral banner.
4. Make sure that clicking on the banner takes you to the Upwork referral landing page.
5. You will get a 404 since the marketing redirects don't work on stage/calypso.live
6. After the domian the url should be: `/experts/upwork?source=stat-banner`

Testing Tracks events:

You can turn on analytics logging by running this in the console: localStorage.setItem( 'debug', 'calypso:analytics:*' )

- Refresh after running code to active analytics logging
- On viewing the banner: `calypso:analytics:tracks Record event "calypso_upwork_stats_nudge_view" called with props {site_type: "jetpack"}`
- On clicking CTA: `calypso:analytics:tracks Recording event "calypso_upwork_stats_nudge_start_now_button_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`
- On dismissing: `calypso:analytics:tracks Recording event "calypso_upworks_stats_nudge_dismiss_icon_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`

cc @p3ob7o @pcdonohue
